exonum/src/runtime/mod.rs

```diff
@@ -12,77 +12,85 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Transactions Runtime.
-//!
-//! This module contains common building blocks for creating runtimes for the Exonum blockchain.
+//! Common building blocks for creating runtimes for the Exonum blockchain.
 //!
 //! Each runtime contains specific services to execute transactions, process events,
 //! provide user APIs, etc. There is a unified dispatcher that redirects all the calls
-//! and requests to the appropriate runtime environment. Thus, blockchain interacts with the
-//! dispatcher, and not with a specific runtime instance.
+//! and requests to the appropriate runtime environment. Thus, a blockchain interacts with the
+//! dispatcher, and not with specific runtime instances.
+//!
+//! # Artifacts
 //!
-//! # Service Life Cycle
+//! Each runtime has its own [artifacts] registry from which users can create service.
+//! The artifact identifier is required by the runtime for constructing service instances.
+//! In other words, an artifact identifier is similar to a class name, and a specific
+//! service instance - to a class instance. A single artifact may be used to instantiate
+//! zero or more services.
 //!
-//! 1. Each runtime has its own [artifacts] registry from which users can deploy them. The artifact
-//! identifier is required by the runtime for constructing service instances. In other words,
-//! an artifact identifier is similar to a class name, and a specific
-//! service instance - to a class instance.
+//! The format of an artifact ID is uniform across runtimes (it is essentially a string),
+//! but the runtime may customize artifact deployment via a runtime-specific deployment arguments.
 //!
-//! 2. Each validator administrator requests the dispatcher to deploy an artifact
-//! and then the validator node should send the confirmation if this request is successful. Then, if the
-//! number of confirmations is equal to the total number of validators, each validator calls the
-//! dispatcher to register the artifact as deployed. After that the validators can send requests to
-//! start new service instances from this artifact.
+//! # Service lifecycle
 //!
-//! 3. To start a new service instance, each validator administrator should send a request
-//! to the dispatcher. Each request contains exactly the same artifact identifier, instance name, and
-//! instance configuration parameters. Then, as in the previous case, if the number of
-//! confirmations is equal to the total number of validators, each validator calls the dispatcher
-//! to start a new service instance.
+//! 1. An artifact with the service is deployed on the blockchain. The decision to deploy an
```

Comment:
Maybe it makes sense to add a "0." note explaining where deployable artifacts come from (short & abstract)?